### PR TITLE
Corresponding component lib date picker changes

### PR DESCRIPTION
**Pull request by vonbearshark**
## Background
ECE client tweak corresponding to [the component lib date picker changes](https://github.com/ctoec/component-library/pull/141)

## GitHub Issue
#1006 

## [Validation Plan](https://github.com/ctoec/component-library/pull/141#issuecomment-820653718)
 - [x] Confirm the datepicker component works when editing a child's date of birth
 - [x] Confirm the datepicker component works when updating an enrollment's start date
 - [x] Confirm the datepicker component works when updating an enrollment's end date
 - [x] Confirm the datepicker component works when updating the determination date for a family income
 - [x] Confirm the datepicker component works when using the Previous month filter within the Roster view
 
## Automated Testing
- [x] All unit tests are passing.
- [x] All e2e tests are passing.

## Additional Context
`yarn` changes included here to help validation. Will be updated to component lib version bump once those changes are ready.

### CODE REVIEW

**Comment by vonbearshark** [comment]
Ug, yep can confirm some wonky behavior in other `DateInput` usages throughout the app.

For example I can't filter by month in the Roster...

**Comment by vonbearshark** [comment]
This should also fix the minor ux issues in the roster refactor @jlhogan noticed re: starting in an error state and also triggering an `Invalid` date selection

**Review by JordanGuinn — APPROVED** [review]

**Comment by vonbearshark** [comment]
Okay, I made some tweaks to both PRs, did my own validation, and feel like they are both ready for final review/validation. Going to deploy to QA so folks can poke it themselves. Once the corresponding PR is merged in I will change the dependency to just bump the version rather than pulling from gh

**Comment by vonbearshark** [comment]
deployed to QA

Snapshots have changed, and tests might be wonky, so I might keep updating this branch. But should be good to review / validate independently, especially since I need to swap the dependency stuff once the component lib changes make it in. So there will be another chance for 👁️ after

File: client/package.json [hunk 1]
```diff
@@ -47,7 +47,7 @@
     "build": "react-scripts build",
     "test": "TZ=America/New_York CI=true react-scripts test --env=jest-environment-jsdom-sixteen --silent --testPathIgnorePatterns ./*/integrationTests",
     "test-integration": "TZ=America/New_York CI=true react-scripts test --env=jest-environment-jsdom-sixteen --silent --runInBand --testTimeout=50000 --testPathPattern ./*/integrationTests",
-    "test-local": "TZ=America/New_York CI=true react-scripts test --env=jest-environment-jsdom-sixteen",
+    "test-local": "TZ=America/New_York CI=true react-scripts test --env=jest-environment-jsdom-sixteen --testPathIgnorePatterns ./*/integrationTests",
```
Comment (vonbearshark) [inline client/package.json:50]: sorry this is unrelated. happy to rm from this PR, but I think this is accurate otherwise?

Comment (JordanGuinn) [reply client/package.json:50]: Agreed - think this is probably what's desired anyway!

**Review by bamader — APPROVED** [review]
Code looks fine to me, going manual validation now

**Comment by bamader** [comment]
Validation Plan: I'll be testing basically every date picker I can think of in the app in three ways--1) can open date picker and select a date, 2) the selected date causes the expected effect (i.e. changing the value in a form field, or changing a filter or something), 3) the selected date can be saved and retrieved correctly.

NOTE: I'm not going to test the create flow version of date pickers for two reasons: 1) the create flow is broken like 50% of the time anyway so that's not a productive use of debugging time, 2) the create flow should _in theory_ be retrieving the same form page and elements as those contained in EditRecord, so hopefully testing there (where the app won't crash) can at least litmus test the behavior in create.

- [ ] Roster month filter
- [ ] EditRecord birthdate
- [ ] EditRecord determination date
- [ ] Income determination date can clear/unclear depending on not disclosed
- [ ] Enrollment start date selected from Edit Enrollment
- [ ] Enrollment start date selected from Change Enrollment
- [ ] Enrollment end date selected from Change Enrollment
- [ ] Enrollment end date selected from Edit Enrollment
- [ ] Enrollment end date selected from Withdraw Enrollment
- [ ] Funding end date selected from Change Funding
- [ ] Funding end date selected from Edit Funding
- [ ] Funding start date selected from Add Funding
- [ ] Funding start selected from Edit Funding

**Review by vonbearshark — COMMENTED** [review]
Should probably double-check my thinking on the `snap` diffs. They are pretty huge diffs, and at every point the component is used throughout the app. But I _think_ it's what we want? I am pretty sure the rendering of the expanded calendar was just a little wonky and also that this one doesn't show at default?

**Comment by vonbearshark** [comment]
Local tests are failing on `audit`. Seems like there is a security issue? _Maybe_ from the new `carbon` component, but I don't really see how. We didn't actually _add_ carbon this time around--it was already here

**Comment by vonbearshark** [comment]
e2e tests are rightfully broken right now. Not actual issues seemingly, but looking for DOM elements in the date picker that no longer exist. Other tests look good locally, though

**Comment by JordanGuinn** [comment]
Agreed on the snapshot bit - looks like it's just a code reduction in all the areas that the datepicker is used, so I'm not worried about it.

**Comment by vonbearshark** [comment]
Code seems good to me, fwiw. I can't actually approve, though, since it was opened as my PR

**Comment by bamader** [comment]
Validation Plan: I'll be testing basically every date picker I can think of in the app in three ways--1) can open date picker and select a date, 2) the selected date causes the expected effect (i.e. changing the value in a form field, or changing a filter or something), 3) the selected date can be saved and retrieved correctly.

NOTE: I'm not going to test the create flow version of date pickers for two reasons: 1) the create flow is broken like 50% of the time anyway so that's not a productive use of debugging time, 2) the create flow should _in theory_ be retrieving the same form page and elements as those contained in EditRecord, so hopefully testing there (where the app won't crash) can at least litmus test the behavior in create.

- [x] Roster month filter
- [x] EditRecord birthdate
- [x] EditRecord determination date
- [x] Income determination date can clear/unclear depending on not disclosed
- [x] Enrollment start date selected from Edit Enrollment
- [x] Enrollment start date selected from Change Enrollment
- [x] Enrollment end date selected from Change Enrollment
- [x] Enrollment end date selected from Edit Enrollment
- [ ] Enrollment end date selected from Withdraw Enrollment
- [ ] Funding end date selected from Change Funding
- [ ] Funding end date selected from Edit Funding
- [ ] Funding start date selected from Add Funding
- [ ] Funding start selected from Edit Funding

File: e2e-tests/utils/deleteChildInRoster.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+const { scrollToElement } = require('./scrollToElement');
```
Comment (vonbearshark) [inline e2e-tests/utils/deleteChildInRoster.js:1]: Are we still using this file?

File: e2e-tests/utils/enterFormValue.js [hunk 3]
```diff
@@ -7,6 +7,14 @@ module.exports = {
     await browser.clearValue(...selectorArgs);
     await browser.setValue(...selectorArgs, input);
     await browser.pause(1000); // Wait for change... annoying but better than flaky tests
+
+    //  Hacks to click somewhere else on the page to hide the calendar display
```
Comment (vonbearshark) [inline e2e-tests/utils/enterFormValue.js:11]: I guess this makes sense. Might want to add a `TODO` to get rid of this eventually

File: e2e-tests/utils/deleteChildInRoster.js (hunk 2 above)
Comment (nickbristow) [reply e2e-tests/utils/deleteChildInRoster.js:1]: we aren't, but could be useful in the future

**Review by JordanGuinn — APPROVED** [review]

**Comment by JordanGuinn** [comment]
@bamader Thanks for validating this!!  In response to your findings:

- Birth date formatting glitch: Honestly, I think it's probably fine for now as-is, and can be something we track/address in the future if it comes up.
- Birth date datepicker doesn't include a calendar popup: This is intended, as it matches current behavior!
- Clearing/unclearing Income determination fields: That sounds like a related but distinct bug with those other income determination fields.  I confirmed in staging that if you select then deselect `Income Not Disclosed`, then save again, your household size and annual household income values are erased, even though they don't _appear to be_ in the UI.  We should make a bug for that for sure, but independently of this work.

Everything else was green otherwise, though?!

**Comment by bamader** [comment]
Yes everything else was green for me! Just wanted to verify those caveats and extra bugs (we should probs also make tickets for those reporting period and validation/banner alert bugs, yeah?), but it sounds like they're working as intended for now. I give this the seal of approval 🐳

**Comment by JordanGuinn** [comment]
Woot!  And yes - we'll get tickets created for all those issues
